packages/deb/hazelcast/DEBIAN/control

```diff
@@ -5,7 +5,7 @@ Section: imdg
 Priority: optional
 Architecture: all
 Conflicts: ${CONFLICTS}
-Depends: default-jdk-headless | java8-sdk-headless
+Depends: default-jdk-headless | java21-sdk-headless
```

Comment:
```suggestion
Depends: java21-sdk-headless
```

I'm pretty sure we should not have `default-jdk-headless` as alternative for `java21-sdk-headless` as it can be lower then 21.

Tested:
```shell
docker run -it --rm ubuntu bash -c "apt update && apt install -y default-jdk-headless && java -version"
```

FYI @frant-hartm

Reply:
Good spot, I didn't realise that was what [it was doing](https://www.debian.org/doc/debian-policy/ch-relationships.html#syntax-of-relationship-fields)!